modules/speech-to-text-impl/src/main/java/org/opencastproject/speechtotext/impl/engine/WhisperCppEngine.java

```diff
@@ -0,0 +1,417 @@
+    if (whispercppSplitOnWord.isSome()) {
+      logger.debug("WhisperC++ split on word set to {}", whispercppSplitOnWord);
+    }
+
+    whispercppBestOf = OsgiUtil.getOptCfgAsInt(cc.getProperties(), WHISPERCPP_BEST_OF_CONFIG_KEY);
+    if (whispercppBestOf.isSome()) {
+      logger.debug("WhisperC++ best of set to {}", whispercppBestOf);
+    }
+
+    whispercppWordThreshold = OsgiUtil.getOptCfg(cc.getProperties(), WHISPERCPP_WORD_THRESHOLD_CONFIG_KEY).bind(
+        Strings.toDouble);
+    if (whispercppWordThreshold.isSome()) {
+      logger.debug("WhisperC++ word threshold set to {}", whispercppWordThreshold);
+    }
+
+    whispercppEntropyThreshold = OsgiUtil.getOptCfg(cc.getProperties(), WHISPERCPP_ENTROPY_THRESHOLD_CONFIG_KEY).bind(
+        Strings.toDouble);
+    if (whispercppEntropyThreshold.isSome()) {
+      logger.debug("WhisperC++ entropy threshold set to {}", whispercppEntropyThreshold);
+    }
+
+    whispercppLogProbThreshold = OsgiUtil.getOptCfg(cc.getProperties(), WHISPERCPP_LOG_PROB_THRESHOLD_CONFIG_KEY).bind(
+        Strings.toDouble);
+    if (whispercppLogProbThreshold.isSome()) {
+      logger.debug("WhisperC++ log prob threshold set to {}", whispercppLogProbThreshold);
+    }
+
+    whispercppDiarization = OsgiUtil.getOptCfgAsBoolean(cc.getProperties(), WHISPERCPP_DIARIZATION_CONFIG_KEY);
+    if (whispercppDiarization.isSome()) {
+      logger.debug("WhisperC++ diarization set to {}", whispercppDiarization);
+    }
+
+    whispercppTinyDiarization = OsgiUtil.getOptCfgAsBoolean(cc.getProperties(), WHISPERCPP_TINY_DIARIZATION_CONFIG_KEY);
+    if (whispercppTinyDiarization.isSome()) {
+      logger.debug("WhisperC++ tiny diarization set to {}", whispercppTinyDiarization);
+    }
+
+    whispercppNoFallback = OsgiUtil.getOptCfgAsBoolean(cc.getProperties(), WHISPERCPP_NO_FALLBACK_CONFIG_KEY);
+    if (whispercppNoFallback.isSome()) {
+      logger.debug("WhisperC++ no fallback set to {}", whispercppNoFallback);
+    }
+
+    logger.debug("Finished activating/updating speech-to-text service");
+  }
+
+  //TODO: Add method for language detection
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.speechtotext.api.SpeechToTextEngine#generateSubtitlesFile(File, File, String, Boolean)
+   */
+
+  @Override
+  public Map<String, Object> generateSubtitlesFile(File mediaFile,
+      File preparedOutputFile, String language, Boolean translate)
+          throws SpeechToTextEngineException {
+
+    if (!FilenameUtils.isExtension(mediaFile.getPath(), "wav")) {
+      throw new SpeechToTextEngineException("WhisperC++ currently doesn't support any media extension other than wav");
+    }
+
+    String[] baseCommands = { whispercppExecutable,
+        mediaFile.getAbsolutePath(),
+        "--model", whispercppModel,
+        "-ovtt",
+        "-oj",
+        "--output-file", preparedOutputFile.getAbsolutePath().replaceFirst("[.][^.]+$", "")};
+
+    List<String> command = new ArrayList<>(Arrays.asList(baseCommands));
```

Comment:
Instead of creating an array, converting that to a list which is then used to initialize an `ArrayList` below, you could directly work with a list. Something like:
```suggestion
    List<String> baseCommand = List.of(whispercppExecutable,
        mediaFile.getAbsolutePath(),
        "--model", whispercppModel,
        "-ovtt",
        "-oj",
        "--output-file", preparedOutputFile.getAbsolutePath().replaceFirst("[.][^.]+$", ""));

    List<String> command = new ArrayList<>(baseCommand);
```